public/app/plugins/datasource/elasticsearch/datasource.js

```diff
@@ -207,7 +207,7 @@ function (angular, _, moment, kbn, ElasticQueryBuilder, IndexPattern, ElasticRes
         var luceneQuery = angular.toJson(target.query || '*');
         // remove inner quotes
         luceneQuery = luceneQuery.substr(1, luceneQuery.length - 2);
-        luceneQuery = luceneQuery.replace(" and ", " AND ").replace(" or "," OR ").replace(" not "," NOT ");
+        luceneQuery = luceneQuery.replace(/( and )(?=(?:(?:[^"]*"){2})*[^"]*$)/g, ' AND ').replace(/( or )(?=(?:(?:[^"]*"){2})*[^"]*$)/g, ' OR ').replace(/( not )(?=(?:(?:[^"]*"){2})*[^"]*$)/g, ' NOT ');
```

Comment:
what is the bug that is being fixed?
not able to open https://jira.mypaytm.com/secure/browse/AR-5954

Reply:
https://jira.mypaytm.com/browse/AR-5954